lib/config.coffee

```diff
@@ -151,32 +153,9 @@ exports.validate = (config) ->
 	if disallowedProperty?
 		throw new Error("Validation: #{disallowedProperty} not recognized")
 
-###*
-# @summary Get a device configuration object from an application
-# @public
-# @function
-#
-# @param {String} application - application name
-# @param {Object} [options={}] - options
-# @param {String} [options.wifiSsid] - wifi ssid
-# @param {String} [options.wifiKey] - wifi key
-#
-# @returns {Promise<Object>} device configuration
-#
-# @todo Move this to the SDK
-#
-# @example
-# deviceConfig.getByApplication 'App1',
-# 	network: 'wifi'
-# 	wifiSsid: 'foobar'
-# 	wifiKey: 'hello'
-# .then (configuration) ->
-# 	console.log(configuration)
-###
-exports.getByApplication = (application, options = {}) ->
+getByApplication = (application, options = {}) ->
 	Promise.props
 		application: resin.models.application.get(application)
-		apiKey: resin.models.application.getApiKey(application)
```

Comment:
A `deviceApiKey` only makes sense/means anything when coupled with the `uuid` of a registered device, otherwise it's completely useless.  So my reasoning is it only makes sense to allow a `deviceApiKey` to be passed when you're going by device, and in fact it should be required if you're generating for a specific device going forwards.  I was planning on having back-compat in each module to avoid having to do a major bump straight away, but having thought more on it I think it makes sense to major bump these modules and have backwards compat only in the final sdk/cli layer